test_common.py

```diff
@@ -0,0 +1,219 @@
+#   - UNSUPERVISED_METRICS: all unsupervised cluster metrics
+#
+# Those dictionaries will be used to test systematically some invariance
+# properties, e.g. invariance toward several input layout.
+#
+
+#metrics used to test similarity between bicluster
+BICLUSTER_METRICS = {
+    "jaccard":_jaccard ,
+    "consensus_score":consensus_score    
+    }
+      
+SUPERVISED_METRICS = {
+    "adjusted_mutual_info_score":adjusted_mutual_info_score ,
+    "adjusted_rand_score":adjusted_rand_score,
+    "completeness_score":completeness_score,
+    "contingency_matrix":contingency_matrix,
+    "homogeneity_score":homogeneity_score,
+    "entropy":entropy,
+    "expected_mutual_information":expected_mutual_information,
+    "homogeneity_completeness_v_measure":homogeneity_completeness_v_measure,
+    "mutual_info_score":mutual_info_score,
+    "normalized_mutual_info_score":normalized_mutual_info_score,
+    "v_measure_score":v_measure_score
+    }
+        
+UNSUPERVISED_METRICS={
+    "silhouette_score":silhouette_score,
+    "silhouette_samples":silhouette_samples
+    }
+        
+ALL_METRICS = dict()
+ALL_METRICS.update(BICLUSTER_METRICS)
+ALL_METRICS.update(SUPERVISED_METRICS)
+ALL_METRICS.update(UNSUPERVISED_METRICS)
+
+# Lists of metrics with common properties
+# ---------------------------------------
+# Lists of metrics with common properties are used to test systematically some
+# functionalities and invariance, e.g. SYMMETRIC_METRICS lists all metrics that
+# are symmetric with respect to their input argument y_true and y_pred.
+#
+
+# Symmetric with respect to their input arguments y_true and y_pred
+# metric(y_true, y_pred) == metric(y_pred, y_true).
+SYMMETRIC_METRICS = [ "adjusted_rand_score" , "v_measure_score" ,
+    "mutual_info_score","adjusted_mutual_info_score",
+    "normalized_mutual_info_score",
+    ]
+                    
+NON_SYMMETRIC_METRICS = ["homogeneity_score" , "completeness_score"]
+                    
+#METIRICS with output between 0 and 1 and belong to BICLUSTER METRICS
+METRICS_NORMALIZED_OUTPUT_BICLUSTER = ["consensus_score","jaccard"]
+ 
+#metrics with output between 0 and 1                                      
+METRICS_NORMALIZED_OUTPUT = ["adjusted_rand_score","homogeneity_score",
+    "completeness_score","v_measure_score",
+    "adjusted_mutual_info_score","normalized_mutual_info_score"
+    ]                  
+                                
+#when information is zero these metrics output zero
+METRICS_ZERO_INFO = ["normalized_mutual_info_score","v_measure_score",
+    "adjusted_mutual_info_score"
+    ]
+                      
+#METRICS where permutations oflabels dont change score
+METRICS_PERMUTE_LABELS = ["homogeneity_score","v_measure_score",
+    "completeness_score","mutual_info_score","adjusted_mutual_info_score",
+    "normalized_mutual_info_score"
+    ]
+                        
+#metrics which result in 0 when a class is split across different clusters

```

Comment:
It is basically for supervised metrics which give a high score if the clusters are both homogeneous and complete